### PR TITLE
adding support for calibration queries

### PR DESCRIPTION
**Pull request by olyoberdorf**
This is an API equivalent to the "associated calibrations" tab on the Gemini Archive website.  It uses comparable searches to the observation data and returns results one would see on that tab.

**query_calibrations_for_region** - This allows you to get calibrations against a cone search at some coordinate

**query_calibrations_for_object** - This gets calibrations related to a search by object name

**query_calibrations_for_criteria** - This allows for a richer query with multiple terms and returns the calibrations

**query_calibrations_raw** -  For expert use, this allows more free-form search terms based on a URL from the website

Note this pull request depends on BeautifulSoup.  It's already an astropy dependency, but I wasn't sure if there's a desire to avoid it.

### CODE REVIEW

**Comment by bsipocz** [comment]
> Note this pull request depends on BeautifulSoup. It's already an astropy dependency, but I wasn't sure if there's a desire to avoid it.

Using any of the already required dependencies is fair game and encouraged (https://astroquery.readthedocs.io/en/latest/#requirements). If for some reason a supported version needs to be bumped to a newer minimal, that could very much be OK, but needs discussion (we tend not to drop support unless it comes up as a blocker). The least preferable thing, adding any new dependencies should certainly need a prior discussion as in many times it may be possible to work around with the current dependencies.

File: astroquery/gemini/core.py [hunk 1]
```diff
@@ -462,6 +730,76 @@ def _gemini_json_to_table(json):
...
+    response : `~astropy.table.Table`
+    """
+
+    data_table = Table(masked=True)
+
+    table = BeautifulSoup(html, features="html.parser").find("table")
+    # This is future-proofing, the existing page has a bug putting the header inside the body
+    # but if that gets fixed, we'll want this
+    cols = None
+    table_header = table.find("thead")
```
Comment (bsipocz) [inline astroquery/gemini/core.py:754]: This has to handle `NoneType` as an option for `table`, at least the test failure seems legit to me.

**Review by bsipocz — CHANGES_REQUESTED** [review]
Overall this looks good to me, except the test failures are real as some corner cases should be treated in the code. I'll come back for a more thorough review once the CI is all happy.

**Comment by bsipocz** [comment]
Adding some narrative docs would be welcome, too. Please note though that some changes to the gemini docs was merged very recently (now we test the docs examples), so this would need a rebase to pick up those changes.

And I forgot to mention the changelog, please add an entry there, too. Thanks!

File: astroquery/utils/commons.py [hunk 2]
```diff
@@ -23,7 +23,7 @@
 from astropy.coordinates import BaseCoordinateFrame
 
 from ..exceptions import TimeoutError, InputWarning
-from .. import version
+from .. import __version__ as version
```
Comment (bsipocz) [inline astroquery/utils/commons.py:26]: please remove this change

File: astroquery/gemini/core.py [hunk 3]
```diff
@@ -419,6 +422,271 @@ def query_raw(self, *args, **kwargs):
...
+            `~astropy.units` may also be used. Defaults to 0.3 deg.
+
+        Returns
+        -------
+        response : `~astropy.table.Table`
+        """
+        return self.query_calibrations_for_criteria(objectname=objectname, radius=radius)
+
+    @class_or_instance
+    def query_calibrations_for_criteria(self, *rawqueryargs, coordinates=None, radius=None, pi_name=None, program_id=None, utc_date=None,
```
Comment (bsipocz) [inline astroquery/gemini/core.py:477]: what are the possible rawqueryargs and rawquerykwargs possibilities? We preferably should not have those wildcard type options.

**Review by bsipocz — CHANGES_REQUESTED** [review]
going through the code I wonder whether these should be separate methods, or rather a new kwarg for the existing `query_region`, `query_object` etc. 

Also, test failures are very much related and has to be fixed before a proper round of review.

File: astroquery/utils/commons.py (hunk 2 above)
Comment (olyoberdorf) [reply astroquery/utils/commons.py]: oops, done

File: astroquery/gemini/core.py (hunk 3 above)
Comment (olyoberdorf) [reply astroquery/gemini/core.py:477]: The calls are really mapping to a URL search on the Gemini archive website.  These consist of an arbitrary set of arguments in the URL path or key=value arguments as well.  It's order independent on the website.

In the case of `query_raw` and `query_calibrations_raw` I really prefer this open-ended approach so the API mirrors what the web API is doing.  For the `_criteria` calls I could drop them.  They are just a convenience if you want to use the more well defined functions but still slip in one or two unusual criteria or terms.
